applications/tari_validator_node/src/p2p/services/template_manager/manager.rs

```diff
@@ -108,6 +108,10 @@ impl TemplateManager {
             .get_template(address)?
             .ok_or(TemplateManagerError::TemplateNotFound { address: *address })?;
 
+        if matches!(template.status, TemplateStatus::Invalid) {
+            return Err(TemplateManagerError::TemplateCodeHashMismatch);
+        }
```

Comment:
We should only return ok if the template is available for use. Let's create a new error variant for this case.

```suggestion
        if !matches!(template.status, TemplateStatus::Active | TemplateStatus::Deprecated) {
            return Err(TemplateManagerError::TemplateUnavailable);
        }
```